### PR TITLE
New Proposed Pull Request Template

### PR DESCRIPTION
**Pull request by manojvazirani**
The following updates contain the new proposed Pull request template for engineering playbook. This Pull Request template can be used by engineering in different engagements with customers. We can change and use this template based on the requirements from the project. It has impact on the current projects that they are used in.

It starts with few questions that act as pre-requisites before opening a Pull Request.

It has the following sections relevant to a PR:
- Link to the design document for the current PR
- Current Behavior
- New Behavior
- Logs, Error Output, etc
- Other information or known dependencies.

This template is relevant for any customer engagements as well as open source projects.
@SaraSp

### CODE REVIEW

File: Engineering/Templates/PullRequestTemplate.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Thank you for contributing.
+Before submitting this PR, please make sure:
+
+-------------------------------------
+* [ ] Have I followed the guidelines in Contributing as Developers?
```
Comment (davidkpiano) [inline Engineering/Templates/PullRequestTemplate.md]:
```suggestion
* [ ] I followed the guidelines in [Contributing as Developers](https://github.com/microsoft/code-with-engineering-playbook/blob/master/CONTRIBUTING.md)
```

File: Engineering/Templates/PullRequestTemplate.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+* [ ] Have I added an explanation of what my changes do and why I would like you to include them?
```
Comment (davidkpiano) [inline Engineering/Templates/PullRequestTemplate.md]:
```suggestion
* [ ] I added an explanation of what my changes do and why I would like you to include them
```

File: Engineering/Templates/PullRequestTemplate.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* [ ] I have updated the documentation accordingly.
+* [ ] I have added tests to cover my changes.
+* [ ] All new and existing tests passed.
+* [ ] My code follows the code style of this project.
+* [ ] I ran lint checks locally prior to submission.
+* [ ] Have I checked to ensure there aren't other open Pull Requests for the same update/change?
```
Comment (davidkpiano) [inline Engineering/Templates/PullRequestTemplate.md]:
```suggestion
* [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change
```

File: Engineering/Templates/PullRequestTemplate.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+* [ ] Does this introduce a breaking change?
```
Comment (davidkpiano) [inline Engineering/Templates/PullRequestTemplate.md]:
```suggestion
* [ ] This does not introduce a breaking change
```

File: Engineering/Templates/PullRequestTemplate.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+## Link to the design document that prompted this PR?
+-------------------------------------
+<!-- Please provide a ink to the design document relevant to this area -->
```
Comment (davidkpiano) [inline Engineering/Templates/PullRequestTemplate.md]:
```suggestion
<!-- Please provide a link to the design document relevant to this area -->
```

File: Engineering/Templates/PullRequestTemplate.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+-------------------------------------
+<!-- Please describe the behavior or changes that are being added by this PR. -->
+
+-
+
+<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->
+
+## Any relevant logs, error output, etc?
+-------------------------------------
+If it’s long, please upload to (StorageAccount)/pr-support/attachments/(PR Number)/(yourFiles) using (Azure Storage Explorer or portal.azure.com) and insert the link here.
```
Comment (davidkpiano) [inline Engineering/Templates/PullRequestTemplate.md]:
```suggestion
If it’s long, please upload to (StorageAccount)/pr-support/attachments/(PR Number)/(yourFiles) using [Azure Storage Explorer](https://azure.microsoft.com/en-us/features/storage-explorer/) or [portal.azure.com](portal.azure.com) and insert the link here.
```

**Comment by manojvazirani** [comment]
Thanks @davidkpiano for your review ad suggestions.

File: Engineering/Templates/PullRequestTemplate.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
+Thank you for contributing.
+Before submitting this PR, please make sure:
+
+-------------------------------------
+* [ ] I have followed the guidelines in [Contributing as Developers](https://github.com/microsoft/code-with-engineering-playbook/blob/master/CONTRIBUTING.md)
+* [ ] I have added an explanation of what my changes do and why I would like you to include them
+* [ ] I have updated the documentation accordingly.
```
Comment (KSLHacks) [inline Engineering/Templates/PullRequestTemplate.md]: Lines 7-11. Inconsistent line endings (periods). I'd recommend including periods on all items in the checklist.

If for some reason we decide no periods on checklist items, we should remove for consistency.

File: Engineering/Templates/PullRequestTemplate.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+* [ ] I have added tests to cover my changes.
+* [ ] All new and existing tests passed.
+* [ ] My code follows the code style of this project.
+* [ ] I ran lint checks locally prior to submission.
+* [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change
+* [ ] This does not introduce a breaking change
+
+## Link to the design document that prompted this PR?
+-------------------------------------
+<!-- Please provide a link to the design document relevant to this area -->
```
Comment (KSLHacks) [inline Engineering/Templates/PullRequestTemplate.md]: Missing period to be consistent with other comment statements.

**Comment by fnocera** [comment]
@manojvazirani could you please update this PR for the new playbook structure? Thank you!

File: pull-requests/PullRequestTemplate.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
+Thank you for contributing.
+Before submitting this PR, please make sure:
+
+-------------------------------------
+* [ ] I have followed the guidelines in [Contributing as Developers](https://github.com/microsoft/code-with-engineering-playbook/blob/master/CONTRIBUTING.md).
+* [ ] I have added an explanation of what my changes do and why I would like you to include them.
```
Comment (fnocera) [inline pull-requests/PullRequestTemplate.md]: Would reword to "I have explained my changes and reasons why they should be included."

File: pull-requests/PullRequestTemplate.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+* [ ] I have updated the documentation accordingly.
+* [ ] I have added tests to cover my changes.
+* [ ] All new and existing tests passed.
+* [ ] My code follows the code style of this project.
+* [ ] I ran lint checks locally prior to submission.
```
Comment (fnocera) [inline pull-requests/PullRequestTemplate.md]: I have run linting checks prior to submission.

File: pull-requests/readme.md [hunk 11]
```diff
@@ -2,4 +2,5 @@
 
 ## Sections within Pull Requests
 
+* [Pull Request Template](./PullRequestTemplate.md)
```
Comment (fnocera) [inline pull-requests/readme.md]: Please rename to lower case eg: pull-request-template.md. And also please add a link to this template from the main pull-request page https://github.com/microsoft/code-with-engineering-playbook/tree/master/pull-requests

File: pull-requests/PullRequestTemplate.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
+Thank you for contributing.
+Before submitting this PR, please make sure:
+
+-------------------------------------
+* [ ] I have followed the guidelines in [Contributing as Developers](https://github.com/microsoft/code-with-engineering-playbook/blob/master/CONTRIBUTING.md).
```
[the remark this answers is not in the document]
Comment (manojvazirani) [reply pull-requests/PullRequestTemplate.md]: This can act as the default contributing guide and once the project has their own, the PR template for the project can be replaced specific to the project.
